app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/DefaultSettingsRepository.kt

```diff
@@ -44,6 +50,22 @@ class DefaultSettingsRepository @Inject constructor(
         return true
     }
 
+    override fun getSafeFactoryAddress(): BigInteger {
+        return (preferences.getString(PREF_KEY_SAFE_FACTORY_ADDRESS, null) ?: BuildConfig.SAFE_FACTORY_ADDRESS).hexAsEthereumAddress()
+    }
+
+    @Throws(InvalidAddressException::class)
+    override fun setSafeFactoryAddress(address: String?) {
+        val checkAddress = if (address.isNullOrBlank() ||
+                address?.removePrefix("0x") == BuildConfig.SAFE_FACTORY_ADDRESS.removePrefix("0x")) null else address
+        if (checkAddress?.isValidEthereumAddress() == false) {
+            throw InvalidAddressException()
+        }
+        preferences.edit {
+            putString(PREF_KEY_SAFE_FACTORY_ADDRESS, checkAddress)
+        }
```

Comment:
For simplicity we can could remove some checks (ie.: `isValidEthereumAddress` seems enough and we can always update if it's valid)

```kotlin
    @Throws(InvalidAddressException::class)
    override fun setSafeFactoryAddress(address: String?) {
        if (address?.isValidEthereumAddress() == false) {
            throw InvalidAddressException()
        }
        preferences.edit { putString(PREF_KEY_SAFE_FACTORY_ADDRESS, address) }
    }
```

Reply:
the first "check" is for resetting the default value ... so not a real check. Or did I misunderstand?